internal/testtools/diag.go

```diff
@@ -57,14 +110,21 @@ func CompareDiags[D diagnostics.Generic](t *testing.T, fm map[string]*hcl.File,
 
 func compareDiags(t *testing.T, fm map[string]*hcl.File, diags diagnostics.Diag, asserts [][]Assert) {
 	t.Helper()
-	if !matchBiject(diags, asserts) {
-		var buf strings.Builder
-		diagnostics.PrintDiags(&buf, diags, fm, false)
-		t.Fatalf("\n\n%s", buf.String())
+	if !matchBiject(t, diags, asserts) {
+		var b strings.Builder
+		b.WriteString("Actual diagnostics:\n")
+		for _, diag := range diags {
+			b.WriteString("\n")
+			b.WriteString(fmt.Sprintf("[Severity]: %s\n", severityToString(diag.Severity)))
+			b.WriteString(fmt.Sprintf("[Summary]: %s\n", diag.Summary))
+			b.WriteString(fmt.Sprintf("[Details]: %s\n", diag.Detail))
+			b.WriteString("\n")
+		}
+		t.Fatal(b.String())
```

Comment:
@Andrew-Morozko I found this output easier to read when debugging tests, but maybe there are edge cases of `compareDiags` usage I've missed!

Reply:
Yep, I see a problem with this approach: in your case the order of the asserts must match the order of diagnostics. I made this assert system so it doesn't care about the order of the diagnostics coming in. 

I expect that the diagnostic order will be most of the time the same, but parallel code might sometimes emmit them in a different order, creating a semi-broken test. And besideds, I don't think that *order* of the diagnostics shouldn't matter, only that they match the given diags.

Regarding improvements to the output, I've repeated your work at the exact same time 😅. If you want to change something else - I thing #180 could be the place

Reply:
go ahead, Andrew! I'll rebase this on #180